prometheus.lua

```diff
@@ -669,7 +680,7 @@ end
 -- Log an error, incrementing the error counter.
 function Prometheus:log_error(...)
   ngx.log(ngx.ERR, ...)
-  self.dict:incr(ERROR_METRIC_NAME, 1, 0)
+  self.errors:inc(1)
```

Comment:
I am curious to know the reason you needed to change this (and make related changes in `init` and `init_worker`). `ERROR_METRIC_NAME` is deliberately not using per-worker counters to make error reporting more robust, and also testing easier (I believe you needed to add a bunch of `sync()` calls in tests after this change).

Reply:
The reason for this was to ensure that errors are correctly processed in `key_index`. The original call would bypass the `key_index:add` call and in the end it would not be listed in `collect`.

However, it should be possible to add it into the `key_index` explicitly in `init` and than just use the dict directly. It'll be more verbose, but it should satisfy your requirement of robustness.

Reply:
Done in 3486a6252dff71dcec9f4b2f7e43ce4b0af49153